domrl/engine/state_view.py

```diff
@@ -24,6 +24,20 @@ def __init__(self, player, is_player: bool):
         self.all_cards = [card for card in player.discard_pile] + \
             [card for card in player.hand] + \
             [card for card in player.draw_pile]
+        self.all_card_names = [card.name for card in player.discard_pile] + \
+            [card.name for card in player.hand] + \
+            [card.name for card in player.draw_pile]
+        def CountFrequency(my_list):
```

Comment:
from collections import Counter
Counter(list) returns a dictionary that holds the same info as freq on line 37

Reply:
i agree